scripts/globals/survival_guide.lua

```diff
@@ -0,0 +1,200 @@
+require("scripts/globals/settings")
+require("scripts/globals/survival_guide_map")
+
+dsp = dsp or {}
+dsp.survivalGuide = dsp.survivalGuide or {}
+
+-- Determines if the survival guide teleport cost is like if you had a Rhapsody in White key item. Does not affect UI! (Default: 0)
+local SURVIVAL_GUIDE_TELEPORT_COST_GIL = 1000
+local SURVIVAL_GUIDE_TELEPORT_COST_TABS = 50
+
+-- This is used for the NationTeleport save/get
+local travelType = dsp.teleport.type.SURVIVAL
+local cutsceneID = 8500
+
+local optionMap =
+{
+    TELEPORT = 1,
+    UNKNOWN = 2,
+    SET_MENU_LAYOUT = 3,
+    ADD_FAVORITE = 4,
+    REMOVE_FAVORITE = 5,
+    REPLACE_FAVORITE = 6,
+    TELEPORT_MENU = 7
+}
+
+-------------------------------------------------
+-- local functions
+-------------------------------------------------
+
+-- 1 = Original zones
+-- 2 = Zilart Zones
+-- 4 = Chains of Promethia Zones
+-- 8 = Treasures of Aht Urgahn
+-- 16 = Wings of the Goddess
+-- 2048 = Seekers of Auldoin
+local function getEnabledExpansions()
+    -- Original areas and zilart are always available.
+    local returnValue = 3
+
+    if ENABLE_COP == 1 then returnValue = returnValue + 4 end
+
+    if ENABLE_TOAU == 1 then returnValue = returnValue + 8 end
+
+    if ENABLE_WOTG == 1 then returnValue = returnValue + 16 end
+
+    if ENABLE_SOA == 1 then returnValue = returnValue + 2048 end
+
+    return returnValue
+end
+
+local function checkForRegisteredSurvivalGuide(player, guide)
+    local group = guide.group
+    local mask = nil
+    local variableSuffix = ''
+    local hasRegisteredGuide = player:hasTeleport(travelType, guide.groupIndex - 1, guide.group - 1)
+
+    if not hasRegisteredGuide then
+        player:messageSpecial(zones[guide.zoneId].text.COMMON_SENSE_SURVIVAL)
+        player:addTeleport(travelType, guide.groupIndex - 1, guide.group - 1)
+
+        return false
+    end
+
+    return true
+end
+
+local function teleportMenuUpdate(player, option)
+    local choice = bit.band(option, 0xFF)
+
+    if choice >= optionMap.SET_MENU_LAYOUT and choice <= optionMap.TELEPORT_MENU then
+        local favorites = player:getTeleportMenu(travelType)
+
+        local index = bit.rshift(bit.lshift(option, 8), 24)
```

Comment:
Is this equivalent to `bit.rshift(option, 16)`?

Reply:
Fixed

Reply:
It's not the same, it was correct before.

Reply:
Was the idea to give the option padding bits on the right?

Reply:
Lua bit ops return signed 32 bit values. The lshift truncates the upper byte 
so the next byte gets extracted by the shift right.

e.g.
```
bit.lshift(0x01020304, 8)
0x02030400
bit.rshift(0x02030400, 24)
0x02
```
i.e.
`bit.rshift(bit.lshift(option, 8), 24)`
is the same as
`bit.rshift(bit.band(option, 0xFF0000), 16)`